sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -259,6 +259,12 @@ case class CachedRDDBuilder(
       rowCountStats.add(batch.numRows)
       batch
     }.persist(storageLevel)
```

Comment:
I see the problem now. Ideally, we should only use the accumulators to get the stats if the RDD is fully cached. However, RDD cache is lazy, and I don't see an easy way to check the caching status.

One idea is, we call `BlockManager.getStatus` to see if the first partition of this RDD is cached, which should probably work in most cases.

That said, let's update the `isCachedColumnBuffersLoaded` method in this class, to really check if the cache is loaded.

Reply:
I thought `BlockManager.getStatus` was expensive. The docs are also indicating that it should mainly be used for testing. Is it ok to do so in `isCachedColumnBuffersLoaded`? 
My concern with only checking one partition of the RDD is that it's still possible to run into this problem if the partition is small and finishes soon before the other partitions. Then, it is still possible to run into this problem.

Reply:
If we want to be conservative we can check all the partitions. I think it's still much cheaper than doing `rdd.count`.

Reply:
Ok, I changed `isCachedColumnBuffersLoaded` to check the block status for all the partitions.